public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -583,7 +583,8 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
       map((res) => {
         const er = new ElasticResponse(sentTargets, res);
 
-        if (sentTargets.some((target) => target.isLogsQuery)) {
+        // TODO: This needs to be revisited, it seems wroong to process ALL the sent queries as logs if only one of them was a log query
```

Comment:
```suggestion
        // TODO: This needs to be revisited, it seems wrong to process ALL the sent queries as logs if only one of them was a log query
```

Reply:
I assume it really felt wrooooooong in my mind 😅 thanks for catching it :)